### PR TITLE
Improve typings for Enumerable

### PR DESCRIPTION
**Pull request by mfeckie**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- Fixes typo where `Enumerable` was `Enumberable`
- Converts return type from `T[]` to `Enumerable<T>` where appropriate to allow for chaining multiple enumerable transforms
- Updates findBy to all `any` type for comparison value

### CODE REVIEW

File: types/ember/index.d.ts [hunk 1]
```diff
@@ -1173,7 +1174,7 @@ declare module 'ember' {
              * can pass an optional second argument with the target value. Otherwise
              * this will match any property that evaluates to `true`.
              */
-            findBy(key: string, value: string): T | undefined;
+            findBy(key: string, value: any): T | undefined;
```
Comment (chriskrycho) [inline types/ember/index.d.ts]: Are the docs wrong here? They say this is of type `string`: [method doc](https://emberjs.com/api/ember/2.16/classes/Enumerable/methods/findBy?anchor=findBy).

Comment (chriskrycho) [reply types/ember/index.d.ts]: Follow-up: the docs appear to be correct; [`findBy` delegates to the `iter` function](https://github.com/emberjs/ember.js/blob/v2.16.2/packages/ember-runtime/lib/mixins/enumerable.js#L550-L552), which [does a lookup with a key](https://github.com/emberjs/ember.js/blob/v2.16.2/packages/ember-runtime/lib/mixins/enumerable.js#L48-L54).

**Comment by dwickern** [comment]
I had to stare at the diff for a while before I saw that typo 😆 

`map`, `filter`, `filterBy`, etc actually return a `NativeArray` which is `T[] & Enumerable<T>`
https://github.com/emberjs/ember.js/blob/v2.16.2/packages/ember-runtime/lib/mixins/enumerable.js

File: types/ember/index.d.ts (hunk 1 above)
Comment (mfeckie) [reply types/ember/index.d.ts]: I'm confused too now!  As I understand it, findBy

1. Iterates through the collection
2. Uses `key` to retrieve a value from the collection
3. if a `value` is provided the two are compared using `===`, so value could be of `any` type

So, I think the type should actually be `value?: any`

example

image

**Comment by mhegazy** [comment]
@chriskrycho can i get another review from you on this?

**Comment by chriskrycho** [comment]
I’ll take a look first thing tomorrow!

**Review by chriskrycho — APPROVED** [review]

**Comment by chriskrycho** [comment]
Looks good here, @mhegazy (and I added the GH review to that effect as well).

**Pull request merged**